### ISSUE TITLE
Character Size

### ISSUE THREAD

**Issue by zem-sessions** [shudnal/NomapPrinter#3]
blaxxun — Today at 12:51 PM
that seems to be map data for the no map printer mod.
does that make up for most of the file?
if you remove the line, does the file get a lot smaller?
obviously remove it from a backup.
like, copy the save file, delete the line in the copy, save the file.

Cass / Zemeck — Today at 12:53 PM
Deleting that line reduces 2.7mb to 36kb.

KG — Today at 1:00 PM
Ask mod author to cache images into folder instead of character save file

---

The map data is currently inflating the character file size which is causing issues with ServerCharacters. I realize this was listed as an untested part on your mod page, but bringing it up in the event you can potentially fix it =D. I've attached a sample character that exhibits the issue.
[Steam_76561197985890896_tarsius.zip](https://github.com/shudnal/NomapPrinter/files/13208377/Steam_76561197985890896_tarsius.zip)

---

edit: GraveBear — Today at 4:48 PM
nah it's good, one person contacting them is good, tell them to join here if they need anything

(here being Odin's Plus discord)

**Comment by shudnal** [comment]
Thanks. I got it. Will make it an option where to save the data.

**Comment by shudnal** [comment]
Implemented in 1.0.12.
Added new option to save/load map data to local file instead of character save file.
That option is not server synced as players should set the file name theirselves.
If it is really great inconvinience to make players set it pls open new issue and I'll find some time on it later.

**Comment by zem-sessions** [comment]
This is great, appreciate it. Should be able to pre-set that kind of thing with modpacks so it's not an issue that it doesn't sync imo.

**Comment by shudnal** [comment]
I've come to the proper serversynced solution. Previous had bad implementation. Pls check version 1.0.13.
Now you can set option "Store map in local folder" as relative path (something like "screenshots\nomapdata") and in that case it would be qualified as subfolder for Valheim local save folder. For Windows it's ..\AppData\LocalLow\IronGate\Valheim.
So if you set "Store map in local folder" to "screenshots/nomapdata" then file for pair player-world will be saved to 
..\AppData\LocalLow\IronGate\Valheim\screenshots\nomapdata\shudnal.NomapPrinter.PlayerName.WorldName.png
on Player.Save and then loaded from that file on Player.Load.

If you set option "Store map in local folder" as full qualified folder name then the files will be saved there without path calculations.

Option "Store map in local folder" is server synced and have less priority than "Load map from file" shared serverside map option.

**Comment by shudnal** [comment]
That's best solution I can see now for multiple servers and different settings. It should be enough in your case without extra actions.

**Comment by zem-sessions** [comment]
WorldName: Testing

Produces error on connect/disconnect. Occurs whether character is new or has already touched a table.

### On Connect:
`[Info   :Nomap Printer] Can't find file (C:/Users/Robert/AppData/LocalLow/IronGate/Valheim\1\shudnal.NomapPrinter.Localtest.Testing.png)!`

### On Disconnect:
```
[Info   :Nomap Printer] Saving nomap data to C:/Users/Robert/AppData/LocalLow/IronGate/Valheim\1\shudnal.NomapPrinter.Localtest.Testing.png
[Info   :Nomap Printer] Saving map to local file error:
System.IO.DirectoryNotFoundException: Could not find a part of the path "C:\Users\Robert\AppData\LocalLow\IronGate\Valheim\1\shudnal.NomapPrinter.Localtest.Testing.png".
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00164] in <695d1cc93cca45069c528c15c9fdd749>:0
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int)
  at System.IO.File.Create (System.String path, System.Int32 bufferSize) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0
  at System.IO.File.Create (System.String path) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0
  at System.IO.File.WriteAllBytes (System.String path, System.Byte[] bytes) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0
  at NomapPrinter.NomapPrinter.SaveMapToLocalFile (Player player) [0x0003a] in <bca95c58b3c442ccb3910fc14aba2231>:0
```

Character interacts with cartography table:
```
[Info   :Nomap Printer] Saves going to C:/Users/Robert/AppData/LocalLow/IronGate/Valheim\screenshots
[Info   :Nomap Printer] Saves going to C:/Users/Robert/AppData/LocalLow/IronGate/Valheim\screenshots
[Info   :Nomap Printer] Finished Map Draw
```

Config:
[shudnal.NomapPrinter.cfg.zip](https://github.com/shudnal/NomapPrinter/files/13257947/shudnal.NomapPrinter.cfg.zip)

**Comment by shudnal** [comment]
You didn't create a folder for a file. Fixed in 1.0.14
